static/app/views/releases/thresholdsList/index.tsx

```diff
@@ -48,51 +55,58 @@ function ReleaseThresholdList({}: Props) {
-  //       /**
-  //        * Include environments from:
-  //        * all projects if the user is a superuser
-  //        * the requested projects
-  //        * all member projects if 'my projects' (empty list) is selected.
-  //        * all projects if -1 is the only selected project.
-  //        */
-  //       if (
-  //         (selectedProjects.length === 1 &&
-  //           selectedProjects[0] === ALL_ACCESS_PROJECTS &&
-  //           project.hasAccess) ||
-  //         (selectedProjects.length === 0 && (project.isMember || user.isSuperuser)) ||
-  //         selectedProjects.includes(project.id)
-  //       ) {
-  //         return project.environments;
-  //       }
-
-  //       return [];
-  //     })
-  //   );
-  //   const envDiff = new Set([...allEnvSet].filter(x => !unSortedEnvs.has(x)));
-
-  //   return Array.from(unSortedEnvs)
-  //     .sort()
-  //     .concat([...envDiff].sort());
-  // };
-
-  // NOTE: currently no way to filter for 'None' environments
+  const selectedProjects: Project[] = useMemo(() => {
+    return projects.filter(project =>
+      selection.projects.some(id => String(id) === project.id || id === -1)
+    );
+  }, [projects, selection.projects]);
+
+  const getAllEnvironments = (): string[] => {
+    const selectedProjectIds = selection.projects.map(id => String(id));
+    const {user} = ConfigStore.getState();
+    const allEnvSet = new Set(projects.flatMap(project => project.environments));
+    // NOTE: mostly taken from environmentSelector.tsx
+    const unSortedEnvs = new Set(
+      projects.flatMap(project => {
+        /**
+         * Include environments from:
+         * all projects if the user is a superuser
+         * the requested projects
+         * all member projects if 'my projects' (empty list) is selected.
+         * all projects if -1 is the only selected project.
+         */
+        if (
+          (selectedProjectIds.length === 1 &&
+            selectedProjectIds[0] === String(ALL_ACCESS_PROJECTS) &&
+            project.hasAccess) ||
+          (selectedProjectIds.length === 0 && (project.isMember || user.isSuperuser)) ||
+          selectedProjectIds.includes(project.id)
+        ) {
+          return project.environments;
+        }
+
+        return [];
+      })
+    );
+    const envDiff = new Set([...allEnvSet].filter(x => !unSortedEnvs.has(x)));
+
+    // bubble the selected projects envs first, then concat the rest of the envs
+    return Array.from(unSortedEnvs)
+      .sort()
+      .concat([...envDiff].sort());
+  };
+
+  /**
+   * Thresholds filtered by environment selection
+   * NOTE: currently no way to filter for 'None' environments
+   */
   const filteredThresholds = selection.environments.length
-    ? thresholds.filter(
-        threshold => selection.environments.indexOf(threshold.environment.name) > -1
-      )
+    ? thresholds.filter(threshold => {
+        const envName = threshold.environment ? threshold.environment.name : '';
```

Comment:
'' is not really being used. How about to turn this into a one liner?
`
return threshold.environment?.name ? selection.environments.indexOf(envName) > -1 : !selection.environments.length
`

My point is envName could just be `threshold.environment?.name` this if is not doing much.

Reply:
Ah that's a good callout 👍